pkg/archive/archive_test.go

```diff
@@ -256,12 +259,34 @@ func testArchive(t *testing.T, when spec.G, it spec.S) {
 				tr := tar.NewReader(file)
 
 				verify := h.NewTarVerifier(t, tr, 1234, 2345)
-				verify.NextDirectory("/nested/dir/dir-in-archive", int64(fs.ModePerm&^archive.Umask))
-				verify.NextFile("/nested/dir/dir-in-archive/some-file.txt", "some-content", fileMode(t, filepath.Join(src, "some-file.txt")))
-				verify.NextDirectory("/nested/dir/dir-in-archive/sub-dir", fileMode(t, filepath.Join(src, "sub-dir")))
-				if runtime.GOOS != "windows" {
-					verify.NextSymLink("/nested/dir/dir-in-archive/sub-dir/link-file", "../some-file.txt")
-				}
+				verify.NextDirectory("/nested/dir/dir-in-archive", int64(os.ModePerm))
+			})
+			when("mode is set to -1", func() {
+				it("writes a tar to the dest dir with default (0777) dir mode", func() {
+					fh, err := os.Create(filepath.Join(tmpDir, "some.tar"))
+					h.AssertNil(t, err)
+
+					tw := tar.NewWriter(fh)
+
+					err = archive.WriteDirToTar(tw, src, "/nested/dir/dir-in-archive", 1234, 2345, -1, true, true, nil)
+					h.AssertNil(t, err)
+					h.AssertNil(t, tw.Close())
+					h.AssertNil(t, fh.Close())
+
+					file, err := os.Open(filepath.Join(tmpDir, "some.tar"))
+					h.AssertNil(t, err)
+					defer file.Close()
+
+					tr := tar.NewReader(file)
+
+					verify := h.NewTarVerifier(t, tr, 1234, 2345)
+					verify.NextDirectory("/nested/dir/dir-in-archive", 0777)
+					verify.NextFile("/nested/dir/dir-in-archive/some-file.txt", "some-content", 0777)
+					verify.NextDirectory("/nested/dir/dir-in-archive/sub-dir", 0777)
+					if runtime.GOOS != "windows" {
+						verify.NextSymLink("/nested/dir/dir-in-archive/sub-dir/link-file", "../some-file.txt")
+					}
```

Comment:
We should keep the old assertions, as the child entries should still have their original permissions, it's just the root directory that should get 0777

Reply:
I have to change the implementation because I assumed we wanted to give the children those permissions too.
Thank you, Natalie, I'll fix it 😁

Reply:
Now it should have been fixed :)